subprojects/gstreamer/libs/gst/base/gstbaseidlesrc.c

```diff
@@ -1192,36 +1311,61 @@ gst_base_idle_src_add_timestamp (GstBaseIdleSrc * src, GstBuffer * buf)
   GST_BUFFER_DTS (buf) = running_time;
 }
 
+static gboolean
+gst_base_idle_src_buffer_list_add_timestamp_func (GstBuffer ** buf, guint idx,
+    gpointer user_data)
+{
+  GstBaseIdleSrc *src = GST_BASE_IDLE_SRC (user_data);
+  gst_base_idle_src_add_timestamp (src, *buf);
+  return TRUE;
+}
+
 static void
 gst_base_idle_src_process_object (GstBaseIdleSrc * src, GstMiniObject * obj)
 {
+  GstFlowReturn ret = GST_FLOW_OK;
   GstPad *pad = src->srcpad;
 
   GST_PAD_STREAM_LOCK (pad);
 
   if (GST_IS_BUFFER (obj)) {
     GstBuffer *buf = GST_BUFFER_CAST (obj);
-    GstFlowReturn ret;
 
     if (src->priv->do_timestamp) {
       gst_base_idle_src_add_timestamp (src, buf);
     }
 
     GST_DEBUG_OBJECT (src, "About to push Buffer %" GST_PTR_FORMAT, buf);
-
     ret = gst_pad_push (pad, buf);
-    if (ret != GST_FLOW_OK)
-      GST_ERROR ("Got ret: %s", gst_flow_get_name (ret));
+    goto check_ret_error;
+  } else if (GST_IS_BUFFER_LIST (obj)) {
+    GstBufferList *buf_list = GST_BUFFER_LIST_CAST (obj);
+
+    if (src->priv->do_timestamp) {
+      gst_buffer_list_foreach (buf_list,
+          gst_base_idle_src_buffer_list_add_timestamp_func, src);
+    }
+
+    GST_DEBUG_OBJECT (src, "About to push BufferList %" GST_PTR_FORMAT,
+        buf_list);
+    ret = gst_pad_push_list (pad, buf_list);
+    goto check_ret_error;
   } else if (GST_IS_EVENT (obj)) {
     GstEvent *event = GST_EVENT_CAST (obj);
-    gboolean ret;
-
     GST_DEBUG_OBJECT (src, "About to push Event %" GST_PTR_FORMAT, event);
-    ret = gst_pad_push_event (pad, event);
-    if (!ret)
-      GST_ERROR ("HUUUUAA");
-  } else if (GST_IS_CAPS (obj)) {
-    GST_DEBUG_OBJECT (src, "About to push Caps %" GST_PTR_FORMAT, obj);
+    if (!gst_pad_push_event (pad, event)) {
+      /* TODO: What is the right thing to do here?! */
+      ret = GST_FLOW_CUSTOM_ERROR;
+    }
+    goto check_ret_error;
+  } else {
+    GST_ERROR_OBJECT (src, "Unknown object %" GST_PTR_FORMAT " type", obj);
+  }
+
+check_ret_error:
+  if (ret != GST_FLOW_OK && ret != GST_FLOW_FLUSHING) {
+    // XXX: Should we proxy this into the bus in case or error or just ignore?
+    GST_ERROR ("Got ret: %s", gst_flow_get_name (ret));
```

Comment:
Current push errors are "ignored". As in, we just log an error. 
Should we be posting a message in the bus as the base src does for flow errors?